File: backend/src/main/java/com/parking/smartparking/dto/response/AuthResponse.java
package com.parking.smartparking.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO trả về sau khi Đăng nhập/Đăng ký thành công
 *
 * Response JSON mẫu: { "userId": 1, "fullName": "Nguyen Van A", "email":
 * "user@example.com", "role": "ROLE_USER", "message": "Đăng nhập thành công!" }
 *
 * ⚠️ LƯU Ý: KHÔNG trả về password trong response
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {

    private Long userId;
    private String fullName;
    private String email;
    private String role;

    // [FIX 2 - JWT] Token để Frontend lưu vào localStorage và gửi trong Authorization header
    private String token;

    private String message;
}

File: mobile/smartparking_mobile/lib/core/models/auth_response.dart
class AuthResponse {
  final int userId;
  final String fullName;
  final String email;
  final String role;
  final String? branchCode;
  final String token;

  AuthResponse({
    required this.userId,
    required this.fullName,
    required this.email,
    required this.role,
    required this.token,
    this.branchCode,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      userId: (json['userId'] as num).toInt(),
      fullName: (json['fullName'] ?? '') as String,
      email: (json['email'] ?? '') as String,
      role: (json['role'] ?? 'ROLE_USER') as String,
      branchCode: json['branchCode'] as String?,
      token: (json['token'] ?? '') as String,
    );
  }
}
